src/app/delegation_verify/delegation_verify.ml

```diff
@@ -0,0 +1,229 @@
+open Mina_base
+open Core
+open Async
+
+let get_filenames =
+  let open In_channel in
+  function
+  | [ "-" ] | [] ->
+      input_all stdin |> String.split_lines
+  | filenames ->
+      filenames
+
+let verify_snark_work ~verify_transaction_snarks ~proof ~message =
+  verify_transaction_snarks [ (proof, message) ]
+
+let config_flag =
+  let open Command.Param in
+  flag "--config-file" ~doc:"FILE config file" (optional string)
+
+let keyspace_flag =
+  let open Command.Param in
+  flag "--keyspace" ~doc:"Name of the Cassandra keyspace" (required string)
+
+let no_checks_flag =
+  let open Command.Param in
+  flag "--no-checks" ~aliases:[ "-no-checks" ]
+    ~doc:"disable all the checks, just extract the info from the submissions"
+    no_arg
+
+let block_dir_flag =
+  let open Command.Param in
+  flag "--block-dir" ~aliases:[ "-block-dir" ]
+    ~doc:"the path to the directory containing blocks for the submission"
+    (required Filename.arg_type)
+
+let cassandra_executable_flag =
+  let open Command.Param in
+  flag "--executable"
+    ~aliases:[ "-executable"; "--cqlsh"; "-cqlsh" ]
+    ~doc:"the path to the cqlsh executable"
+    (optional Filename.arg_type)
+
+let timestamp =
+  let open Command.Param in
+  anon ("timestamp" %: string)
+
+let instantiate_verify_functions ~logger = function
```

Comment:
Note that this will use the constants from the config file correctly, but *won't* know which signature kind to use, and will blindly use the one from the compile config. In particular, building with the standard (`dev`) dune profile will lead to `Testnet` signatures (injected [here](https://github.com/MinaProtocol/mina/blob/a6745833668b5d704d18ff1f50a7d0f8af71dfa4/src/lib/signature_kind/compile_config/mina_signature_kind.ml#L11)). If you would like to build binaries with explicit signature kinds, you can use the trick [here](https://github.com/MinaProtocol/mina/blob/a6745833668b5d704d18ff1f50a7d0f8af71dfa4/src/app/cli/src/dune#L31) to override the signature kind, to have explicit versions of the binaries. For reference, using that override causes us to use a different implementation of the dune virtual module (the one [here](https://github.com/MinaProtocol/mina/blob/develop/src/lib/signature_kind/mainnet/mina_signature_kind.ml)), so that mainnet signatures will be used regardless of what's specified in `config.mlh`.

Reply:
Nice, thanks @mrmr1993.